raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -436,14 +509,49 @@ private FetchQuorumRecordsResponseData buildOutOfRangeFetchQuorumRecordsResponse
             .setHighWatermark(highWatermark.orElse(-1L));
     }
 
+    private CompletableFuture<FetchQuorumRecordsResponseData> handleFetchQuorumRecordsRequest(
+        RaftRequest.Inbound requestMetadata
+    ) throws IOException {
+        FetchQuorumRecordsRequestData request = (FetchQuorumRecordsRequestData) requestMetadata.data;
+        FetchQuorumRecordsResponseData response = tryCompleteFetchQuorumRecordsRequest(request);
+        if (response.errorCode() != Errors.NONE.code() ||
```

Comment:
Should we also consider `min.bytes` here? This could further build up the batch.

Reply:
I think this needs more discussion, but I am inclined not to implement `min.bytes` for metadata fetches. In practice, I think this is rarely used for normal partition replication and I am not sure we ever want to delay metadata propagation. It also has a more complex implementation. With an assumption of `min.bytes=1`, then we can simplify completion logic to just check offsets.